modules.d/90overlay-root/module-setup.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+check() {
+    # do not add modules if the kernel does not have overlayfs support
+    [ -d $dracutsysrootdir/lib/modules/$kernel/kernel/fs/overlayfs ] || return 1
+}
+
+depends() {
+    # We do not depend on any modules - just some root
+    return 0
+}
+
+# called by dracut
+installkernel() {
+    instmods overlay
```

Comment:
Might need `hostonly=''`

Reply:
You  are right, overlayfs and NFSv4 is a little bit tricky.

As of 2019 overlayfs  can only work with NFS v4 if acl(5) support is disabled on the filessystem on the NFS server side. This can be done by remounting the nfsroot on the NFS server with noacl. I'm not sure if newer kernel version still need this.

Reply:
Included. Keeping open because of the unrelated comment.